daemon/logs.go

```diff
@@ -19,10 +20,38 @@ import (
 	"github.com/docker/docker/pkg/stdcopy"
 )
 
+type loggerStore struct {
```

Comment:
This may not be absolutely needed... it really comes down to the logs API endpoint needing to attach to a logger object, which can either be created as needed or use a cached logger.

If we just create loggers for each reader rather than using a cached logger then we will also have to watch for stop notifications while reading logs.

Decided to keep things as they were and keep a cache of loggers rather than creating a new one for each reader.